kits/arm/arm.py

```diff
@@ -0,0 +1,217 @@
+        self.eff_cmd = np.zeros(self.grp.size)
+        
+        # Setup gravity vector for grav comp
+        self.gravity_vec = [0, 0, 1]
+        gravity_from_quaternion(self.fbk.orientation[0], output=self.gravity_vec)
+        self.gravity_vec = np.array(self.gravity_vec)
+        
+        # Zero times
+        self.time_now = time()
+        self.traj_start_time = self.time_now
+        self.duration = 0
+        self.t_traj = 0
+        
+        self.set_goal_first = True
+        self.at_goal = True
+        self.goal = "grav comp"
+        
+    
+    # Loads gains from given gains file    
+    def loadGains(self, gains_file):
+        try:
+            self.cmd.read_gains(gains_file)
+            if not self.grp.send_command_with_acknowledgement(self.cmd):
+                raise RuntimeError('Did not receive ack from group.')
+            print('Successfully read gains from file and sent to module.')
+        except Exception as e:
+            print('Problem reading gains from file or sending to module: {0}'.format(e))
+    
+    
+    # Updates feedback and generates the basic command for this timestep
+    def update(self):
+        self.prev_fbk = list(self.fbk)
+        self.grp.get_next_feedback(reuse_fbk=self.fbk)
+        self.time_now = perf_counter()
+        if not self.at_goal:
+            self.t_traj = self.time_now - self.traj_start_time
+            if self.time_now > (self.traj_start_time + self.duration):
+                self.at_goal = True
+        # Create command
+        if self.goal == "grav comp":
+            # Calculate required torques to negate gravity at current position
+            nan_array = np.empty(self.grp.size)
+            nan_array[:] = np.nan
+            self.cmd.position = nan_array
+            self.cmd.velocity = nan_array
+            self.cmd.effort = get_grav_comp_efforts(self.model, self.fbk.position, -self.gravity_vec)
+        elif not self.at_goal:
+            # Move to target position
+            self.pos_cmd, self.vel_cmd, self.accel_cmd = self.trajectory.get_state(self.t_traj)
+            self.cmd.position = self.pos_cmd
+            self.cmd.velocity = self.vel_cmd
+            self.cmd.effort = get_grav_comp_efforts(self.model, self.fbk.position, -self.gravity_vec)
+        else:
+            # Stay at current position
+            self.cmd.position = self.fbk.position_command
+        return 
+    
+    
+    def send(self):
+        # Send command
+        self.grp.send_command(self.cmd)
+        return
+
+    
+    def createMotionArray(self, size, prev_cmd, array=None, flow=None):
+        # Helper function to create arrays for things like waypoints, velocities, and accelerations that are used in creating a trajectory
+        # Note flow overrides any array passed (used for vel and accel, should never pass flow and positions)
+        mArray = np.empty((self.grp.size, size+1))
+        mArray[:, 0] = prev_cmd
+        if flow == None:
```

Comment:
Pedantic thing: Apparently it's better practice to do `if flow is None:` here instead.

I think the rationale is that `None` is a "singleton" of sorts... There is only one `None` instance in a Python interpreter. That means that if `flow` is equal to `None` (literally, if flow is "null"), it's underlying object is the same object id (_i.e._, both variables point to the same object in memory) as `None`, meaning you can use the `is` operator.